ContextBasedFeature.cpp

```diff
@@ -191,11 +194,12 @@ bool ContextBasedFeature::prepare(	const CorePoints& corePoints,
 			bool cancelled = false;
 #ifndef _DEBUG
 #if defined(_OPENMP)
-			omp_set_num_threads(std::max(1, omp_get_max_threads() - 2));
-#pragma omp parallel for
+#pragma omp parallel for num_threads(omp_get_max_threads())
```

Comment:
don't you still want to reduce the number of threads? (to allow the OS and the GUI to be responsive)